proof/refine/ARM/Finalise_R.thy

```diff
@@ -3225,52 +3222,50 @@ lemma (in delete_one_conc_pre) finaliseCap_replaceable:
-             schedContextUnbindAllTCBs_obj_at'_tcb_None
-           | simp add: isZombie_Null isThreadCap_threadCapRefs_tcbptr
-                       isArchObjectCap_Cap_capCap not_obj_at'
-           | (rule hoare_strengthen_post [OF arch_finaliseCap_removeable[where slot=slot]],
-                  clarsimp simp: isCap_simps)
-           | wpc
-           | strengthen invs_valid_objs' invs_sch_act_wf')+
-  apply clarsimp
+   apply (wpsimp wp: prepares_delete_helper'' [OF cancelAllIPC_unlive]
+                     prepares_delete_helper'' [OF cancelAllSignals_unlive]
+                     unbindMaybeNotification_obj_at'_ntfnBound
+                     unbindMaybeNotification_obj_at'_no_change
+               simp: isZombie_Null)
+    apply (strengthen invs_valid_objs' invs_sch_act_wf')
+    apply (wpsimp wp: schedContextMaybeUnbindNtfn_obj_at'_ntfnSc
+                      prepares_delete_helper'' [OF replyClear_makes_unlive]
+                      hoare_vcg_if_lift_strong simp: isZombie_Null)+
+        apply (clarsimp simp: obj_at'_def)
+       apply (wpsimp wp: schedContextZeroRefillMax_removeable'
+                         prepareThreadDelete_unqueued prepareThreadDelete_nonq
+                         prepareThreadDelete_inactive
+                         suspend_makes_inactive
+                         suspend_nonq
+                         suspend_bound_yt_tcb_at'_None
+                         unbindNotification_bound_tcb_at'
+                         unbindFromSC_bound_sc_tcb_at'_None
+                         schedContextUnbindYieldFrom_makes_unlive
+                         schedContextUnbindReply_obj_at'_reply_None
+                         schedContextUnbindReply_obj_at'_not_reply
+                         schedContextUnbindNtfn_obj_at'_ntfn_None
+                         schedContextUnbindNtfn_obj_at'_not_ntfn
+                         schedContextUnbindAllTCBs_obj_at'_tcb_None
+                   simp: isZombie_Null isThreadCap_threadCapRefs_tcbptr)+
+    apply (rule hoare_strengthen_post [OF arch_finaliseCap_removeable[where slot=slot]],
+           clarsimp simp: isCap_simps)
+   apply (wpsimp wp: deletingIRQHandler_removeable'
+                     deletingIRQHandler_final[where slot=slot])+
   apply (frule cte_wp_at_valid_objs_valid_cap'; clarsimp)
   apply (case_tac "cteCap cte",
          simp_all add: isCap_simps capRange_def cap_has_cleanup'_def
                        final_matters'_def objBits_simps
                        not_Final_removeable finaliseCap_def,
          simp_all add: removeable'_def)
-    (* ThreadCap *)
-    apply (frule capAligned_capUntypedPtr [OF valid_capAligned], simp)
-    apply (clarsimp simp: valid_cap'_def)
-    apply (drule valid_globals_cte_wpD'[rotated], clarsimp)
-    apply (clarsimp simp: invs'_def valid_state'_def valid_pspace'_def)
-   (* ArchObjectCap *)
-   apply (clarsimp simp: obj_at'_def)
-  (* ReplyCap *)
-  apply clarsimp
+       (* ThreadCap *)
```

Comment:
nitpick: I prefer the original indentation of this comment, where it matches the line after it